packages/plugins/bundler-report/src/index.ts

```diff
@@ -29,34 +45,106 @@ const xpackPlugin: (context: GlobalContext) => PluginOptions['webpack'] & Plugin
+                if (options.input) {
+                    if (Array.isArray(options.input)) {
+                        for (const input of options.input) {
+                            directories.add(path.dirname(input));
+                        }
+                    } else if (typeof options.input === 'object') {
+                        for (const input of Object.values(options.input)) {
+                            directories.add(path.dirname(input));
+                        }
+                    } else if (typeof options.input === 'string') {
+                        directories.add(path.dirname(options.input));
+                    } else {
+                        throw new Error('Invalid input type');
+                    }
+                }
+
+                if ('output' in options) {
+                    handleOutputOptions(options.output);
+                }
+            },
+            outputOptions(options) {
+                handleOutputOptions(options);
+            },
+        };
+    };
+
     const bundlerReportPlugin: PluginOptions = {
         name: PLUGIN_NAME,
         enforce: 'pre',
         esbuild: {
             setup(build) {
-                globalContext.bundler.rawConfig = build.initialOptions;
+                context.bundler.rawConfig = build.initialOptions;
 
                 if (build.initialOptions.outdir) {
-                    globalContext.bundler.outDir = build.initialOptions.outdir;
+                    context.bundler.outDir = build.initialOptions.outdir;
                 }
 
                 if (build.initialOptions.outfile) {
-                    globalContext.bundler.outDir = path.dirname(build.initialOptions.outfile);
+                    context.bundler.outDir = path.dirname(build.initialOptions.outfile);
+                }
+
+                if (build.initialOptions.absWorkingDir) {
+                    context.cwd = build.initialOptions.absWorkingDir;
                 }
 
                 // We force esbuild to produce its metafile.
                 build.initialOptions.metafile = true;
             },
         },
-        webpack: xpackPlugin(globalContext),
-        rspack: xpackPlugin(globalContext),
-        // Vite and Rollup have the same API.
-        vite: rollupPlugin(globalContext),
-        rollup: rollupPlugin(globalContext),
+        webpack: xpackPlugin(context),
+        rspack: xpackPlugin(context),
+        // Vite and Rollup have (almost) the same API.
+        // They don't really support the CWD concept,
+        // so we have to compute it based on existing configurations.
+        // The basic idea is to compare input vs output and keep the common part of the paths.
+        vite: {
+            config(config) {
+                if (config.root) {
+                    context.cwd = config.root;
+                } else {
+                    handleCwd(Array.from(directories), context);
+                }
+            },
+            ...rollupPlugin(),
```

Comment:
You may want to put `...rollupPlugin()` before `config(config) {` to avoid having a field name `config` in `rollupPlugin()` overriding your config